frame/pallet-xcm/src/lib.rs

```diff
@@ -0,0 +1,1786 @@
+            fee_asset_item: u32,
+            maybe_weight_limit: Option<WeightLimit>,
+        ) -> DispatchResult {
+            let origin_location = T::ExecuteXcmOrigin::ensure_origin(origin)?;
+            let dest = (*dest).try_into().map_err(|()| Error::<T>::BadVersion)?;
+            let beneficiary: MultiLocation = (*beneficiary)
+                .try_into()
+                .map_err(|()| Error::<T>::BadVersion)?;
+            let assets: MultiAssets = (*assets).try_into().map_err(|()| Error::<T>::BadVersion)?;
+
+            ensure!(
+                assets.len() <= MAX_ASSETS_FOR_TRANSFER,
+                Error::<T>::TooManyAssets
+            );
+            let value = (origin_location, assets.drain());
+            ensure!(
+                T::XcmReserveTransferFilter::contains(&value),
+                Error::<T>::Filtered
+            );
+            let (origin_location, assets) = value;
+            let ancestry = T::LocationInverter::ancestry();
+            let fees = assets
+                .get(fee_asset_item as usize)
+                .ok_or(Error::<T>::Empty)?
+                .clone()
+                .reanchored(&dest, &ancestry)
+                .map_err(|_| Error::<T>::CannotReanchor)?;
+            let max_assets = assets.len() as u32;
+            let assets: MultiAssets = assets.into();
+            let weight_limit = match maybe_weight_limit {
+                Some(weight_limit) => weight_limit,
+                None => {
+                    let beneficiary = beneficiary.clone();
+                    let fees = fees.clone();
+                    let mut remote_message = Xcm(vec![
+                        WithdrawAsset(assets.clone()),
+                        ClearOrigin,
+                        BuyExecution {
+                            fees,
+                            weight_limit: Limited(0),
+                        },
+                        DepositAsset {
+                            assets: Wild(All),
+                            max_assets,
+                            beneficiary,
+                        },
+                    ]);
+                    // use local weight for remote message and hope for the best.
+                    let remote_weight = T::Weigher::weight(&mut remote_message)
+                        .map_err(|()| Error::<T>::UnweighableMessage)?;
+                    Limited(remote_weight)
+                }
+            };
+            let xcm = Xcm(vec![
+                BuyExecution { fees, weight_limit },
+                DepositAsset {
+                    assets: Wild(All),
+                    max_assets,
+                    beneficiary,
+                },
+            ]);
+            let mut message = Xcm(vec![
+                WithdrawAsset(assets),
+                InitiateReserveWithdraw {
+                    assets: Wild(All),
+                    reserve: dest,
+                    xcm,
+                },
+            ]);
```

Comment:
I checked what each line does and it looks good.
I'm just surprised (same as you) that they haven't added support for this.

Will you also make a PR for `substrate`?

Reply:
Someday lets make PR to polkadot with it.

Reply:
Sure, that sounds good.

If we can avoid maintaining this pallet in our repo, it'd be nice.
I'm pretty sure we can expect heavy changes here as new XCM version roll out.

Reply:
interesting. Maybe they don't want their DOTs back xD